### PR TITLE
Adding BASEFEE Method

### PR DESCRIPTION
**Pull request by nagarev**
Adding `BASEFEE` method

## Description

The [BASEFEE](https://eips.ethereum.org/EIPS/eip-3198) Opcode (EIP-3198) gives the EVM access to the block’s base fee. This PR aims to add support for that method on RSKj.

## Motivation and Context

Maintain compatibility with ETH.

## How Has This Been Tested?

- Local testing
- Automated tests
- QA

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Requires Activation Code (Hard Fork)

* **Other information**:

**REQUIRES HF ACTIVATION**

### CODE REVIEW

**Review by Vovchyk — APPROVED** [review]

File: rskj-core/src/main/java/org/ethereum/vm/VM.java [hunk 1]
```diff
@@ -1834,6 +1847,8 @@ protected void executeOpcode() {
             break;
             case OpCodes.OP_GASLIMIT: doGASLIMIT();
             break;
+            case OpCodes.OP_BASEFEE: doBASEFEE();
```
Comment (Vovchyk) [inline rskj-core/src/main/java/org/ethereum/vm/VM.java:1850]: probably we'd need to add here a check for activation height and throw an exception (`Program.ExceptionHelper.invalidOpCode()`), if it's not activated yet

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/org/ethereum/vm/VM.java (hunk 1 above)
Comment (nagarev) [reply rskj-core/src/main/java/org/ethereum/vm/VM.java:1850]: Yup!! Let's add that!

**Comment by nagarev** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

File: rskj-core/src/test/java/co/rsk/vm/opcode/BasefeeDslTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
+package co.rsk.vm.opcode;
```
Comment (asoto-iov) [inline rskj-core/src/test/java/co/rsk/vm/opcode/BasefeeDslTest.java:1]: not sure if it is needed in test classes too, but copyright header missing here.

**Review by asoto-iov — DISMISSED** [review]

File: rskj-core/src/test/java/co/rsk/vm/opcode/BasefeeDslTest.java (hunk 2 above)
Comment (nagarev) [reply rskj-core/src/test/java/co/rsk/vm/opcode/BasefeeDslTest.java:1]: Totally! Will add it!

**Review by Vovchyk — APPROVED** [review]

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Review by Vovchyk — APPROVED** [review]

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Pull request merged**